packages/api/amplify_api/test/web_socket/web_socket_types_test.dart

```diff
@@ -85,5 +85,26 @@ void main() {
         expect(message.messageType, entry.expectedMessageType);
       });
     }
+
+    test('should decode data errors as a list', () {
+      final errors = [
+        {
+          'message':
+              'Cannot return null for non-nullable type: "AWSDateTime" within parent "Blog" (/onCreateBlog/updatedAt)',
+          'path': ['onCreateBlog', 'updatedAt']
+        }
+      ];
+      final entry = {
+        'id': 'xyz-456',
+        'type': 'data',
+        'payload': {'data': null, 'errors': errors}
+      };
+      final message = WebSocketMessage.fromJson(entry);
+      expect(message.messageType, MessageType.data);
+      expect(
+        message.payload!.toJson()['errors'],
+        errors,
+      );
+    });
   });
```

Comment:
request: Could you add either another test or another assertion here that the graphql response decoder will accept the JSON produced by this websocket message? Basically, something like:

```
final response = GraphQLResponseDecoder.instance.decode<T>(
    request: request, // you will have to add a request for this which can be basic subscription request
    response: message.payload.toJson(),
 );
expect(response.errors.first.message, errorMessage); // should make the errorMessage a const
```

I think part of the issue here isn't just that it's the wrong type, but that we're not matching downstream behavior of how we actually convert the ws message to the stream event we give to the user and the decoder is the part where it will eventually get integrated. Ideally there would also be a test where we mock a message like this on the mocked stream and assert it all works as expected but not required.

Reply:
I agree - it surprised me we didn't have a test which caught this before. No worries at all, though, this is the perfect opportunity to add it 👍

Reply:
Good call, that assertion has been added!